scripts/SREP/get-sosreport/script.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -e
+set -o nounset
+set -o pipefail
+
+## validate input
+if [[ -z "$NODE" ]]
+then
+    echo "Variable node cannot be blank"
+    exit 1
+fi
+
+# Import sftp_upload library
+source /managed-scripts/lib/sftp_upload/lib.sh
+
+# Define expected values
+DUMP_DIR="${PWD}"
+DATE="$(date -u +"%Y%m%dT%H%M")"
+SOSREPORT_FILENAME="sosreport-${NODE}.tar.xz"
+SOSREPORT_FILEPATH="${DUMP_DIR}/sosreport-${NODE}.tar.xz"
+
+check_node(){
+    echo "Checking if \"${NODE}\" is an existing node..."
+    
+    if (oc get nodes --no-headers -oname | grep "${NODE}") &> /dev/null; then
+       echo "[OK] \"${NODE}\" is a node."
+    else
+        echo "[Error] \"${NODE}\" is not a node. Exiting script"
+        exit 1
+    fi
+}
+
+generate_sosreport() {
+# 1st Debug session - Generate the sosreport and keep it in the host
+
+    echo " ==== Generating SOSREPORT ===="
+    oc -n default debug node/"${NODE}" -- sh -c "chroot /host toolbox sos report -k crio.all=on -k crio.logs=on --batch"
+    return 0
+}
+
+validate_file() {
+    printf "==== Check if the sosreport is created inside the node ===="
+    if ! oc -n default debug node/"${NODE}" -- bash -c "ls -lh /host/var/tmp/sosreport-*.tar.xz" | tee;
+        then printf "========\n [Error] Sosreport file not found in node directory /host/var/tmp dir \n========\n"
+        exit 1
+        else printf "========\n [Success] Sosreport file generated\n========\n"
+    fi
+}
+copy_sosreport() {
+# 2nd Debug session - Fetch sosreport .tar.xz file and save in the container volume.
+    printf "\n==== Copying file from node to container volume ====\n"
+    oc -n default debug node/"${NODE}" -- bash -c 'cat $(ls -tA /host/var/tmp/sosreport-*.tar.xz | head -1)' > "${DUMP_DIR}"/sosreport-"${NODE}".tar.xz ;
```

Comment:
```suggestion
    oc -n default debug "node/${NODE}" -- bash -c 'cat $(ls -tA /host/var/tmp/sosreport-*.tar.xz | head -1)' > "${DUMP_DIR}/sosreport-${NODE}.tar.xz" ;
```

You want to double quote things that should result in a single string